src/metabase/upload.clj

```diff
@@ -170,8 +171,45 @@
     "unnamed_column"
     (u/slugify (str/trim raw-name))))
 
+(defn- is-pk?
+  [[type name]]
+  (and (#{"id" "pk" (u/lower-case-en name)})
+       (isa? type ::int)))
+
+(defn- add-pk-column
+  "Adds driver-specific auto-increment and non-nil/PK constraints to the first ID/PK column it finds; otherwise prepends
+  a new ID column to the columns list."
+  [driver name-type-pairs]
+  (let [found-it   (atom false)
+        pked-pairs (map (fn [[name type :as p]]
+                          (if (and (not @found-it)
```

Comment:
:vomiting_face: bu I couldn't figure out how else to do it without doing multiple passes